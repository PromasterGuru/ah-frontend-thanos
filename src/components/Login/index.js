import React from 'react';

const Login = () => (
  <div className="container">
    <form>
      <div className="form-group">
        <label htmlFor="exampleInputEmail1">
          Email address
          <input
            type="email"
            className="form-control"
            id="exampleInputEmail1"
            aria-describedby="emailHelp"
            placeholder="Enter email"
          />
        </label>
      </div>
      <div className="form-group">
        <label htmlFor="exampleInputPassword1">
          Password
          <input
            type="password"
            className="form-control"
            id="exampleInputPassword1"
            placeholder="Password"
          />
        </label>
      </div>
      <button type="submit" className="btn btn-primary">
        Submit
      </button>
    </form>
  </div>
);

export default Login;
