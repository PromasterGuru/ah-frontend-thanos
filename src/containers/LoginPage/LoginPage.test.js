import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Login from '../../components/Login';
import LoginPage from './index';

let store;
const mockStore = configureMockStore([thunk]);
const LoginComponent = shallow(<Login onChange={jest.fn()} onSubmit={jest.fn()} />);
describe('<Login />', () => {
  test('renders the component', () => {
    expect(LoginComponent).toMatchSnapshot();
  });
});

describe('tesing login container', () => {
  const props = {
    signUp: jest.fn(),
    signupData: {
      errors: {
        password: ['Wrong password'],
      },
    },
  };

  let wrapper;
  beforeEach(() => {
    store = mockStore({
      LoginReducer: {
        errors: {
          email: ['The email not valid'],
          password: ['Wrong password'],
        },
      },
    });
  });
  it('should mount view all component', () => {
    wrapper = mount(
      <MemoryRouter>
        <Provider store={store}>
          <LoginPage props={props} loginUser={jest.fn} />
        </Provider>
      </MemoryRouter>,
    );
    expect(wrapper.find('Login').length).toBe(1);
  });
});

describe('handle Invoke email for password reset', () => {
  let component;
  store = mockStore({});
  beforeEach(() => {
    component = mount(
      <MemoryRouter>
        <Provider store={store}>
          <LoginPage />
        </Provider>
      </MemoryRouter>,
    );
  });

  it('should submit user email and password without fail', () => {
    component
      .find('input#email')
      .simulate('change', { target: { name: 'email', value: 'ewrfg' } });
    component
      .find('input#email')
      .simulate('change', { target: { name: 'email', value: '' } });
    component
      .find('input#password')
      .simulate('change', { target: { name: 'password', value: '' } });
    component
      .find('input#password')
      .simulate('change', { target: { name: 'password', value: 'sdfg@dfg' } });

    component.find('form').simulate('submit');
  });
});
