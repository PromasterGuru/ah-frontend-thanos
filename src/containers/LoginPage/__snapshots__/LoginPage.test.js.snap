// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders the component 1`] = `
<div
  className="container"
>
  <div
    className="col-left"
  >
    <img
      alt="Sign up poster"
      src="https://images.pexels.com/photos/34072/pexels-photo.jpg"
    />
  </div>
  <div
    className="col-right"
  >
    <form>
      <h3
        className="text-center"
      >
        Sign In
      </h3>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="email"
          name="email"
          pattern="^[a-zA-Z0-9.!#$%&â€™*+/=?^_\`{|}~-]+@[a-zA-Z0-9-]+(?:.[a-zA-Z0-9-]+)*$"
          placeholder="Email"
          required={true}
          title="Please provide a valid email address"
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <input
          className="form-control"
          id="password"
          name="password"
          placeholder="**********"
          required={true}
          type="password"
        />
      </div>
      <div
        className="signin-buttons"
      >
        <button
          className="btn btn-primary sign-in-bt"
          id="signInBt"
          name="signInBt"
          type="submit"
        >
          Sign In
        </button>
        <div
          className="social-login-icons"
        >
          <small
            className="text-muted p-1"
          >
            Or Sign In with Social Media:
             
          </small>
          <div
            className="icons-group text-center"
          >
            <FontAwesomeIcon
              border={false}
              className="icon"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fab",
                  "facebook-f",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className="icon"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fab",
                  "twitter",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <FontAwesomeIcon
              border={false}
              className="icon"
              fixedWidth={false}
              flip={null}
              icon={
                Array [
                  "fab",
                  "google",
                ]
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>
        </div>
      </div>
      <div
        className="small mt-3"
      >
        <span>
          Not yet a member? 
        </span>
        <a
          href="/signup"
        >
          Sign Up
        </a>
      </div>
    </form>
  </div>
</div>
`;
